tests/test_batch.py

```diff
@@ -217,5 +219,335 @@ def given_configuration_file_version_different_as_river_file_version_when_runnin
+        report = StringIO()
+        reach = "reach"
+        q_location = "location"
+        q_threshold = 0.123
+        q_bankfull = 0.456
+        q_stagnant = 0.789
+        tstag = 0.101112
+        q_fit = [0.131415, 0.171819]
+        Q = [0.232425, 0.262728, 0.293031]
+        T = [0.323334, 0.353637, 0.383940]
+        
+        ApplicationSettingsHelper.PROGTEXTS.clear()
+        
+        dfastmi.batch.write_report(report, reach, q_location, q_threshold, q_bankfull, q_stagnant, tstag, q_fit, Q, T, slength)
+
+        report_lines = report.getvalue().split('\n')
+                
+        prefix = 'No message found for '
+        assert prefix + 'reach' in report_lines
+        assert prefix + 'report_qthreshold' in report_lines
+        assert prefix + 'report_qbankfull' in report_lines
+        assert prefix + 'closed_barriers' in report_lines
+        assert prefix + 'char_discharge' in report_lines
+        assert report_lines.count(prefix + 'char_discharge') is 3
+        assert prefix + 'char_period' in report_lines
+        assert report_lines.count(prefix + 'char_period') is 3
+        assert prefix + 'need_multiple_input' in report_lines
+        assert prefix + 'lowwater' in report_lines
+        assert prefix + 'transition' in report_lines
+        assert prefix + 'highwater' in report_lines
+        assert prefix + 'length_estimate' in report_lines
+        assert prefix + 'prepare_input' in report_lines
+        
+class Test_batch_check_configuration():
+    
+    def given_version_when_check_configuration_then_return_false(self):
+        rivers = RiversObject("dfastmi/Dutch_rivers_v1.ini")
+        config = ConfigParser()
+        
+        assert dfastmi.batch.check_configuration(rivers, config) is False
+        
+    def given_version_with_no_matching_version_when_check_configuration_then_return_false(self):
+        rivers = RiversObject("dfastmi/Dutch_rivers_v1.ini")
+        config = ConfigParser()
         
+        config.add_section("General")
+        config.set("General", "Version", "0.0")
         
+        assert dfastmi.batch.check_configuration(rivers, config) is False
+    
+    class Test_check_configuration_v1():
+        @pytest.fixture
+        def rivers(self):
+            return  RiversObject("dfastmi/Dutch_rivers_v1.ini")
+        
+        @pytest.fixture
+        def config(self):
+            return  ConfigParser()
+                    
+        def given_version_1_when_check_configuration_then_return_false(self, rivers : RiversObject, config : ConfigParser):
+            assert dfastmi.batch.check_configuration(rivers, config) is False
+            
+        def given_general_section_when_check_configuration_then_return_false(self, rivers : RiversObject, config : ConfigParser):
+            self.set_valid_general_section(config, "1.0")
+            
+            assert dfastmi.batch.check_configuration(rivers, config) is False
```

Comment:
use assert not [function_that_returns_bool]

Reply:
is nicer than 
assert [function_that_returns_bool] is False

Reply:
assert not [function_that_returns_bool]
or
assert [function_that_returns_bool] is not True

Reply:
discussed, use assert not [function_that_returns_bool] for false and use assert [function_that_returns_bool] for true